test/metabase/test/gen2.clj

```diff
@@ -0,0 +1,294 @@
+(ns metabase.test.gen2
+  (:require [clojure.spec.alpha :as s]
+            [clojure.test.check.generators :as gen]
+            [clojure.tools.reader.edn :as edn]
+            [java-time :as t]
+            [metabase.models
+             :refer
+             [Activity Card Collection Dashboard DashboardCard DashboardCardSeries Database Dimension Field
+              FieldValues Metric MetricImportantField NativeQuerySnippet PermissionsGroup PermissionsGroupMembership
+              Pulse PulseCard PulseChannel Table User]]
+            [metabase.test :as mt]
+            [metabase.util.date-2.parse :as u.date]
+            [reifyhealth.specmonstah.core :as rs]
+            [reifyhealth.specmonstah.spec-gen :as rsg]
+            [talltale.core :as tt]
+            [toucan.db :as tdb]))
+
+(def ^:private edn-definitions-dir "./test/metabase/test/data/dataset_definitions/")
+
+(defonce ^:private sample-dataset
+  (edn/read-string {:eof nil, :readers {'t #'u.date/parse}}
+                   (slurp (str edn-definitions-dir "sample-dataset.edn"))))
+
+(defn- random-desc
+  "Return a random name from sample-dataset"
+  []
+  (second (rand-nth (nth (first sample-dataset) 2))))
+
+(defn- coin-toss
+  ([] (coin-toss 0.5))
+  ([p] (< (rand) p)))
```

Comment:
This is more of a next iteration thing but https://github.com/incanter/incanter is good for creating random distributions of numbers -- see the sample dataset generation code we linked above